flocker/node/agents/testtools/_blockdevice.py

```diff
@@ -891,117 +891,81 @@ class InvalidConfig(Exception):
 
     :raises: ``InvalidConfig`` if a
         ``FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE`` was not set and the
         default config file could not be read.
 
-    :return: A two-tuple of an ``IBlockDeviceAPI`` implementation and a
-        ``dict`` of keyword arguments that can be used instantiate that
-        implementation.
+    :return: XXX
     """
-    # ie cust0, rackspace, aws
-    platform_name = environ.get('FLOCKER_FUNCTIONAL_TEST_CLOUD_PROVIDER')
-    if platform_name is None:
-        raise InvalidConfig(
-            'Supply the platform on which you are running tests using the '
-            'FLOCKER_FUNCTIONAL_TEST_CLOUD_PROVIDER environment variable.'
+    flocker_functional_test = environ.get('FLOCKER_FUNCTIONAL_TEST')
+    if flocker_functional_test is None:
+        raise SkipTest(
+            'Please set FLOCKER_FUNCTIONAL_TEST environment variable to '
+            'run storage backend functional tests.'
         )
 
     config_file_path = environ.get('FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE')
     if config_file_path is None:
-        raise InvalidConfig(
-            'Supply the path to a cloud credentials file '
+        raise SkipTest(
+            'Supply the path to a backend configuration file '
             'using the FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE environment '
-            'variable. See: '
-            'https://docs.clusterhq.com/en/latest/gettinginvolved/acceptance-testing.html '  # noqa
-            'for details of the expected format.'
+            'variable.'
+        )
+
+    # ie storage-drivers.rackspace
+    config_section = environ.get(
+        'FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_SECTION',
+    )
+    if config_section is None:
+        raise SkipTest(
+            'Supply the section of the config file '
+            'containing the configuration for the driver under test '
+            'with the FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_SECTION '
+            'environment variable.'
         )
 
     with open(config_file_path) as config_file:
         config = yaml.safe_load(config_file.read())
 
-    section = config.get(platform_name)
-    if section is None:
-        raise InvalidConfig(
-            "The requested cloud platform "
-            "was not found in the configuration file. "
-            "Platform: %s, "
-            "Configuration File: %s" % (platform_name, config_file_path)
-        )
-
-    provider_name = section.get('provider', platform_name)
-    try:
-        provider_environment = ProviderType.lookupByName(provider_name)
-    except ValueError:
-        raise InvalidConfig(
-            "Unsupported provider. "
-            "Supplied provider: %s, "
-            "Available providers: %s" % (
-                provider_name,
-                ', '.join(p.name for p in ProviderType.iterconstants())
-            )
-        )
+    section = None
+    for section in config_section.split('.'):
+        config = config.get(section)
```

Comment:
nit, perhaps use `(config or {}).get(section)` or otherwise guard against missing sections before the last one.

specifically the following case:

```
config_section = 'a.b.c.d'
config = {a: {b: {}}}
```
will error out with "NoneType has no attribute get" when section='d' and config equals `None` from the previous attempt at `getting` `c` from `b`.